### DISCUSSION TITLE
Does the use of KenLM affect the license?

### DISCUSSION THREAD

**Discussion by DanielSWolf** [coqui-ai/STT#2229] [General]
The [LICENSE file of Coqui STT](https://github.com/coqui-ai/STT/blob/main/LICENSE) contains the Mozilla Public License (MPL) 2.0. So I was assuming that *the whole* of Coqui STT was under the MPL.

However, the `native_client.tflite.XXX.tar.xz` releases each contain the files `libkenlm.so` and `KenLM_License_Info.txt` with LGPL license information. Am I correct in assuming that any software using Coqui as a library must in turn publish this LGPL license information?

For context: I'm the developer of an MIT-licensed tool. Using an MPL library, though not ideal, would be OK to me, but I was hoping to avoid LGPL libraries.

**Comment by reuben** [comment]
Yes, you must publish the license information for libkenlm.so and maintain dynamic linking to allow users to swap their own implementation. If you have an alternative implementation of optimized LM querying with a more permissive license we're happy to accept a PR!
